cmd/broker/ingress/main.go

```diff
@@ -150,9 +175,29 @@ func (f *Handler) dispatch(msg *provisioners.Message) error {
 type runnableServer struct {
 	logger *zap.Logger
 	s      *http.Server
+	// ShutdownTimeout is the duration to wait for the http.Server to gracefully
+	// shut down when the stop channel is closed. If this is zero, no shutdown
+	// action will be taken.
+	ShutdownTimeout time.Duration
 }
 
-func (r *runnableServer) Start(<-chan struct{}) error {
-	r.logger.Info("Ingress Listening...", zap.String("Address", r.s.Addr))
+func (r *runnableServer) Start(stopCh <-chan struct{}) error {
+	logger := r.logger.With(zap.String("address", r.s.Addr))
+	logger.Info("Listening...")
+	if r.ShutdownTimeout > 0 {
```

Comment:
@Harwayne I moved the shutdown call here so it's easier to run multiple listeners without juggling multiple shutdowns.

Reply:
Makes sense to move it, but to honor the Runnable contract (as I understand it), I want this always shutdown the server. Some values may have no timeout on the shutdown, but I do want Shutdown called.

Reply:
Now Shutdown is always called, and if ShutdownTimeout is not positive, no timeout is given.

An alternate approach that could make sense is:
- Positive: use as timeout
- Zero: use as timeout (shut down immediately)
- Negative: no timeout

Essentially the choice is: should the default behavior be immediate shutdown or infinite shutdown? I'm not sure which is more reasonable, but I'm leaning toward infinite, because problems with infinite shutdown will be obvious (process never stops, users notice downtime) versus subtle for immediate shutdown (transient failure on restart, users can't reproduce).